### ISSUE TITLE
fputcsv() flagged when writing to stream

### ISSUE THREAD

**Issue by paulschreiber** [Automattic/VIP-Coding-Standards#404]
## Bug Description

The below code creates a CSV for download. PHPCS flags this as an error. I'm writing to a stream, not the filesystem.

## Minimal Code Snippet

``` php
$output_stream = fopen( 'php://output', 'w' );

foreach ( $list as $row ) {
    fputcsv( $output_stream, $row );
}
```

## Error Code

```
  88 | ERROR   | Filesystem writes are forbidden, you should not be using
     |         | fputcsv() (WordPress.VIP.FileSystemWritesDisallow)
```

## Environment

| Question                | Answer
| ------------------------| -------
| PHP version             | 7.1.23
| PHP_CodeSniffer version | 3.4.1
| VIPCS version           | 0.2.5

**Comment by GaryJones** [comment]
> VIPCS version | 0.2.5

Does this still happen with VIPCS 0.4.0 (latest published version)?

How about `master`?

**Comment by paulschreiber** [comment]
* Aside: version 0.4.0 of VIPCS conflicts with WPCS 2.1.0.
* Problem still occurs with VIPCS 0.4.0, WPCS 1.2.1,  PHPCS 3.4.1,

**Comment by GaryJones** [comment]
How about `master`?

**Comment by paulschreiber** [comment]
@GaryJones Is there a specific commit or PR you think addresses this or a related problem? Or are you just enjoying having me randomly regress things?

Yes, the problem still occurs with `master` (97fa945).

**Comment by rebeccahum** [comment]
Due to the nature of static analysis, I think that that example would still flagged it because of `$output_stream` being a variable. 

@jrfnl What are your thoughts for this one?

**Comment by jrfnl** [comment]
@rebeccahum  I agree.

Just based on the call to `fputcsv()`, there is no way to know whether the output will be written to a file or a stream.

With the code snippet above, the sniff _could_ walk back to find how `$output_stream` is declared and analyse that, but in reality, most code isn't this clean cut and the `$handle` (`$output_stream` variable) may be passed into a function doing the actual writes, as a function parameter in which case there is no way to know what was passed in.

We know it will be a resource handle, but that's it. Presuming it is a resource handle to `php://output` would be a huge jump and would leave any code which does actually attempt to write to a file undetected, which I think is a worse outcome.
